lib/delta/sink/s3.ex

```diff
@@ -34,6 +35,20 @@ defmodule Delta.Sink.S3 do
     end
   end
 
+  defp apply_filename_rewrite(config, filename) do
+    if Map.has_key?(config, :filename_rewrites) do
+      do_apply_filename_rewrite(config.filename_rewrites, filename)
+    else
+      filename
+    end
+  end
+
+  defp do_apply_filename_rewrite(rewrites, filename) do
+    Enum.reduce(rewrites, filename, fn rewrite, modified_filename ->
+      String.replace(modified_filename, rewrite.pattern, rewrite.replacement)
+    end)
```

Comment:
Should all rewrites be applied in succession like this? It seems like it could be a little tricky—a filename could be transformed by one rewrite into a new one that matches another rewrite later on.

If this is the desired behavior, could you explain it in the line you added to the README? If not, maybe only the first rewrite whose pattern matches the filename should apply?

Reply:
@jzimbel-mbta My intent was for it to apply all of them, but I can see your point regarding confusion. I am open to the idea of it halting after the first match, but I'm leaning towards documenting the existing behavior because these filename rewrites should be kept relatively specific so that kind of stomping is impossible. I'll update the readme, and if this becomes a problem it should be pretty trivial to change 😄